### ISSUE TITLE
Bootsrap is not working when openssl compiled for android

### ISSUE THREAD

**Issue by sl4v3k** [savoirfairelinux/opendht#684]
Hi, just tryed to build "Hello World" application base on libopendht.a taken from android build. 
And i realized that node->bootstrap("bootstap.jami.net", "4222"); is doing anything. 
What i found that bootstrap call chain is finally calling addBootstrap from securedht.h:

    void addBootstrap(const std::string& host, const std::string& service) override {
        logger_->d("DhtRunner::bootstrap from securedht %s %s", host, service); **<----- this is the last log i see**
        dht_->addBootstrap(host, service);
    }

and this is the last log string in logcat. Log put inside dht_->addBootstrap(host, service); is not called:

located in dht.h
    void addBootstrap(const std::string& host, const std::string& service) override {
        logger_->d("dht_->addBootstrap %s %s", host, service); **<------ this is not going to logcat**
        bootstrap_nodes.emplace_back(host, service);
        startBootstrap();
    }

means dht_->addBootstrap(host, service); is calling wrong function.

More, if i compile this on intel x64 all works. Looks like android clang is breaking something? override is not working?

**Comment by sl4v3k** [comment]
Yes You are right i am using DhtProxy client mode, yes make sense local bootstrap is not needed.

I started investigation becuse of android app is loosing connectivity when going to background for some time even with service running and all push enabled.

**Comment by sl4v3k** [comment]
Still simple HW app is not connecting to dht nework, could You check below logs and say what could be possible reason.

com.dht.k0der                        I  [runner 0x7bbb12b020] state changed to Running
com.dht.k0der                        I  [proxy:client] start proxy with dhtproxy.jami.net:80
com.dht.k0der                        I  [proxy:client] starting io_context
com.dht.k0der                        I  [proxy:client] [connectivity] get status
com.dht.k0der                        I  [proxy:client] [info] requesting proxy server node information
com.dht.k0der                        I  [proxy:client] [status] sending request
com.dht.k0der                        I  [proxy:client] [status] query ipv4 info
com.dht.k0der                        I  [proxy:client] [status] query ipv6 info
com.dht.k0der                        I  [http:request:1] connect begin: 51.91.75.152:80 
com.dht.k0der                        I  [connection:1] start http session
com.dht.k0der                        E  [http:request:2] connect: no endpoints provided
com.dht.k0der                        E  [http:request:2] end with error: Transport endpoint is not connected
com.dht.k0der                        E  [proxy:client] [status] ipv6 failed with code=0
com.dht.k0der                        E  [proxy:client] [info] request failed for ipv6
com.dht.k0der                        I  [http:request:1] done with status code 0
com.dht.k0der                        E  [proxy:client] [status] ipv4 failed with code=0
com.dht.k0der                        E  [proxy:client] [info] request failed for ipv4

**Comment by aberaud** [comment]
From the logs, it looks like network requests to the proxy server fail (for both ipv4 and ipv6), which is odd. Was the log captured when the app was in the background ?

Note that to work with push notifications, unfortunately you would need to have a FCM API token configured with the Android app and provided as the "push_token"

**Comment by sl4v3k** [comment]
but all tests done on jami android app with different configurations:
- noPush + service
- withPush FCM / UP + no service
- withPush FCM / UP + service
- all above with and without Proxy

and result same after some time of phone is left no touched jami app stops receiving and sending messages

**Comment by sl4v3k** [comment]
in the logs i see issue related to ICE, mainly:     "All ICE checklists failed" (PJNATH_EICEFAILED)

**Comment by XdekHckr** [comment]
> in the logs i see issue related to ICE, mainly: "All ICE checklists failed" (PJNATH_EICEFAILED). I can get detialed logs if You would like to check it.

Chyba cię szukają: https://x.com/i/status/2069888304334385618

**Comment by sl4v3k** [comment]
co to za ludzie?

**Comment by XdekHckr** [comment]
> co to za ludzie?

Pentesterzy, chyba można tak powiedzieć. Szukają informacji o telefonie zrobionym orzez "k0der"

**Comment by XdekHckr** [comment]
> > co to za ludzie?
> 
> Pentesterzy, chyba można tak powiedzieć. Szukają informacji o telefonie zrobionym orzez "k0der"

A tak w ogóle, to ty jesteś twórcą tego telefonu lub wiesz coś o nim? Bo w sumie zaciekawił mnie ten projekt, a nie ma o nim właściwie nic w polskim internecie...

**Comment by sl4v3k** [comment]
Kim jesteś? Ten telefon to kłopot.

**Comment by XdekHckr** [comment]
> Kim jesteś? Ten telefon to kłopot.

W jakim sensie kłopot? Poza tym po co mam ci mówić kim jestem, skoro nie odpowiedziałeś nawet na moje pytanie. Jestem gapiem z Twittera. XD!

**Comment by sl4v3k** [comment]
nie jestes randomowym gosciem z xa...jedna z kopii offlinowego materialu jest tuaj: https://repower.cc/k0der/
